lib/bolt/config.rb

```diff
@@ -154,12 +181,39 @@ def initialize(boltdir, config_data, overrides = {})
         @transports[key] = transport.default_options
       end
 
+      config_data = merge_config_data(config_data)
+
       update_from_file(config_data)
       apply_overrides(overrides)
 
       validate
     end
 
+    # Merge configuration
+    # Precedence from highest to lowest is: project, user-level, system-wide
+    def merge_config_data(config_data)
+      config_data.each_with_object({}) do |config, acc|
+        @config_files << config[:filepath] if config[:filepath]
+
+        acc.merge!(config[:data]) do |key, val1, val2|
+          case key
+          # Plugin config is shallow merged for each plugin
+          when 'plugins'
+            val1.merge!(val2) { |_, v1, v2| v1.merge(v2) }
+          # Transports are deep merged
+          when 'ssh', 'winrm', 'local', 'remote', 'docker', 'pcp'
+            Bolt::Util.deep_merge(val1, val2)
+          # Hash values are shallow mergeed
+          when 'puppetdb', 'puppetfile_config', 'plugin_hooks', 'apply_settings', 'log'
```

Comment:
The setting in the config file is called `puppetfile`. The variable is `@puppetfile_config` because we already have a `Config#puppetfile` method

Reply:
But [the gist](https://gist.github.com/beechtom/f7b758893796d61a23c8ab868cf8b062) said we were going to overwrite puppetfile settings, not merge them. So in that case, this typo is making it behave correctly.

Reply:
Whoops. As Bob Ross would say - a happy little accident.